src/Domain/CoordinateMaps/KerrHorizon.cpp

```diff
@@ -0,0 +1,177 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/CoordinateMaps/KerrHorizon.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/DereferenceWrapper.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeArray.hpp"
+#include "Utilities/StdArrayHelpers.hpp"
+
+namespace domain::CoordinateMaps {
+
+template <typename T>
+std::array<tt::remove_cvref_wrap_t<T>, 3> KerrHorizon::operator()(
+    const std::array<T, 3>& source_coords) const noexcept {
+  return make_array<tt::remove_cvref_wrap_t<T>, 3>(source_coords) *
+         sqrt(stretch_factor(source_coords));
+}
+
+std::optional<std::array<double, 3>> KerrHorizon::inverse(
+    const std::array<double, 3>& target_coords) const noexcept {
+  const auto coords_mag_sq = dot(target_coords, target_coords);
+  const auto coords_sq_min_spin_sq = coords_mag_sq - spin_mag_sq_;
+  const auto coords_dot_spin = dot(target_coords, spin_);
+  const auto fac = (coords_sq_min_spin_sq +
+                    sqrt(coords_sq_min_spin_sq * coords_sq_min_spin_sq +
+                         4. * square(coords_dot_spin))) /
+                   (2. * coords_mag_sq);
+  // this is the way it was written in spec, but I dont think `fac` can
```

Comment:
@markscheel do you know/remember why this check is in SpEC?

Reply:
I don't remember why it was coded that way in SpEC.  I agree that fac can never be negative (except possibly roundoff).  So what about the case fac==0.0?  That will occur whenever target_coords is orthogonal to spin, and coords_mag_sq is small enough. This is a disk of radius spin_magnitude, oriented normal to the spin.  All target_coords on that disk have fac==0 and therefore map to the origin, so the inverse map is many-to-one.  So perhaps the check should be `fac>0.0` instead of `fac>=0.0` (and same in SpEC as well).

Reply:
@nikwit if you decide to change anything, you may find `Utilities/EqualWithinRoundoff.hpp` useful